core/controllers/reader.py

```diff
@@ -300,7 +300,123 @@ def get(self, exploration_id):
+                issue_customization_args = issue.issue_customization_args
+                # In case issue_keyname is 'state_names', the ordering of the
+                # list is important i.e. [a,b,c] is different from [b,c,a].
+                issue_keyname = stats_models.ISSUE_TYPE_KEYNAME_MAPPING[
+                    issue.issue_type]
+                if (issue_customization_args[issue_keyname] ==
+                        playthrough.issue_customization_args[issue_keyname]):
+                    return index
+        return None
+
+    def _move_playthrough_to_correct_issue(self, playthrough, orig_playthrough):
+        """Moves the updated playthrough to its correct issue in the unresolved
+        issues list.
+
+        Args:
+            playthrough: Playthrough. The updated playthrough domain object.
+            orig_playthrough: Playthrough. The original playthrough domain
+                object which is in the now-incorrect issue list.
+        """
+        max_playthroughs_per_issue = False
+        issue_index = self._find_playthrough_in_exp_issues(playthrough)
+        # Check whether the playthrough can be added to its new issue,
+        # if not, it stays in its old issue.
+        if issue_index is not None:
+            if (len(self.exp_issues.unresolved_issues[
+                    issue_index].playthrough_ids) < (
+                        feconf.MAX_PLAYTHROUGHS_FOR_ISSUE)):
+                self.exp_issues.unresolved_issues[
+                    issue_index].playthrough_ids.append(self.playthrough_id)
+            else:
+                max_playthroughs_per_issue = True
+        else:
+            issue = stats_domain.ExplorationIssue(
+                playthrough.issue_type,
+                playthrough.issue_customization_args,
+                [self.playthrough_id], self.issue_schema_version, is_valid=True)
+            self.exp_issues.unresolved_issues.append(issue)
+
+        # Now, remove the playthrough from its old issue.
+        if not max_playthroughs_per_issue:
+            issue_index = self._find_playthrough_in_exp_issues(orig_playthrough)
+            if issue_index is not None:
+                self.exp_issues.unresolved_issues[
+                    issue_index].playthrough_ids.remove(self.playthrough_id)
+
+    def _assign_playthrough_to_issue(self, playthrough):
+        """Assigns newly created playthrough to its correct issue or makes a new
+        issue.
+
+        Args:
+            playthrough: Playthrough. The playthrough domain object.
+
+        Raises:
+            Exception. Maximum playthroughs per issue reached.
+
+        Returns:
+            playthrough_id: int. The playthrough ID.
+        """
+        # Find whether an issue already exists for the new playthrough.
+        issue_index = self._find_playthrough_in_exp_issues(playthrough)
+        if issue_index is not None:
+            if (len(self.exp_issues.unresolved_issues[
```

Comment:
nit: use temporary variable (`issue = self.exp_issues.unresolved_issues[issue_index]`) to make this line shorter.

Reply:
Done.